lib/chromic_pdf/supervisor.ex

```diff
@@ -393,6 +401,78 @@ defmodule ChromicPDF.Supervisor do
 
       See [`print_header_footer_template.html`](https://cs.chromium.org/chromium/src/components/printing/resources/print_header_footer_template_page.html)
       from the Chromium sources to see how these values are interpreted.
+
+      ### Wait for dynamic content
+
+      If your HTML fetches dynamic content via JavaScript, you can declare the content as "ready" by setting an attribute on a DOM element. You can wait for this attribute with the `wait_for` option.
+
+      The following example waits for the DOM element with ID `#element-id` to receive the attribute `ready-to-print` with value `true`.
+      before printing the page:
+
+          wait_for = %{
+            selector: "#element-id",
+            attribute: "ready-to-print",
+            value: "true"
+          }
+
+          ChromicPDF.print_to_pdf({:url, "http:///example.net"}, wait_for: wait_for)
+
+      The normal timeouts apply.
+
+      #### Example HTML
+
+      The snippet below sets the element once the page has been loaded with this configuration:
+
+          params = [
+            wait_for: %{
+              selector: "#print-ready",
+              attribute: "ready-to-print"
+            }
+          ]
+
+          ChromicPDF.print_to_pdf({:url, "http:///example.net"}, params)
+
+      ```html
+      <!-- Example element that gets dynamic content -->
+      <div id="dynamic"></div>
+      <!-- Marker element that gets the attribute. Can be same as the dynamic element. -->
+      <div id="print-ready"></div>
+
+      <script>
+        function loadDynamicContent() {
+          // Call any JS libraries/frameworks to set the dynamic content.
+          const dynamic_element = document.getElementById('dynamic');
+          dynamic_element.innerText = 'This is filled by JS';
+
+          // When ready, set the attribute on the element you're watching:
+          const element = document.getElementById('print-ready');
+          element.setAttribute('ready-to-print', '');
+        }
+
+        loadDynamicContent();
+      </script>
+      ```
+
+      #### Caveats
+
+      If the attribute already exists on the element the PDF generation will fail with a timeout. This could
+      happen if the attribute is set before `ChromicPDF` has the chance to observe the element. As a workaround
+      you can use `setTimeout` with a small delay. Tune the delay so that it works in your environment.
+
+      ```html
+      <div id="print-ready" />
+
+      <script>
+        window.onload = () => {
+          // Simulate slow content load with 500 ms timeout.
+          setTimeout(() => {
+            const element = document.getElementById('print-ready');
+            element.innerText = "Dynamic content from Javascript";
+            element.setAttribute('ready-to-print', '');
+          }, 500);
+        }
+      </script>
```

Comment:
🤔 

ok for now, I guess. I think I'd like to find a solution for this issue at some point (in other words, runtime conditionals)... I feel this caveat - which likely would go unnoticed by the majority of users - gets too much room in the docs now, and the solution is also quite hack-ish.

Nevertheless, let's leave it as-is for now.

Reply:
Agreed with this being less than stellar hack. Ideally there would be runtime conditionals but it seems to be slightly bigger thing to tackle. I could remove the example from the caveat completely: we can probably assume that people using this library know their way around JS and HTML?

Reply:
as you prefer, yes. I think I'll take a look into runtime conditionals when I find some time and would then hopefully remove this entire section.